frontend/src/store/product/action.js

```diff
@@ -0,0 +1,34 @@
+import actionTypes from './actionTypes';
+import { productService } from '../../services/product-service';
+import { productsActions } from '../../store/products/action';
+
+export const productActions = {
+  createProduct,
+};
+
+function createProduct(title, price, category, image) {
+  return (dispatch) => {
+    dispatch({
+      type: actionTypes.PRODUCT_REQUEST,
+    });
+
+    productService
+      .addProduct({ title, price, category, image })
+      .then(
+        (product) => {
+          dispatch({
+            type: actionTypes.PRODUCT_SUCCESS,
+            product,
+          });
+        },
+        (error) => {
+          alert(error);
+          dispatch({ type: actionTypes.PRODUCT_FAILURE, error });
+        }
+      )
+      .then(() => {
+        const updatedProducts = JSON.parse(localStorage.getItem('products'));
```

Comment:
this should be updated to get products from the backend. Check in with Matt on this